homeassistant/components/websocket_api/connection.py

```diff
@@ -229,7 +229,9 @@ def async_handle(self, msg: JsonValueType) -> None:
     @callback
     def async_handle_close(self) -> None:
         """Handle closing down connection."""
-        for unsub in self.subscriptions.values():
+        # Wrapped in list to avoid
+        # RuntimeError: dictionary changed size during iteration
+        for unsub in list(self.subscriptions.values()):
```

Comment:
This is a bug in production code. Need to make a separate PR

Reply:
This was actually a problem in the mobile_app code. We had two things trying to tear it down at the same time https://github.com/home-assistant/core/pull/115540